File: FleetTrack/src/FleetTrack.Domain/Entities/Driver.cs
using FleetTrack.Domain.Enums;

namespace FleetTrack.Domain.Entities;

public class Driver : BaseEntity
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public string LicenseNumber { get; set; } = string.Empty;
    public DateTime LicenseExpiryDate { get; set; }
    public DriverStatus Status { get; set; }
    public DateTime? LastActiveDate { get; set; }

    // Navigation properties
    public Guid? CurrentVehicleId { get; set; }
    public Vehicle? CurrentVehicle { get; set; }
    public ICollection<Mission> Missions { get; set; } = new List<Mission>();
}

File: fleettrack-frontend/src/types/driver.ts
export enum DriverStatus {
  Available = 0,
  OnDuty = 1,
  OnBreak = 2,
  OffDuty = 3,
  OnLeave = 4,
  Inactive = 5
}

export interface DriverDto {
  id: string;
  firstName: string;
  lastName: string;
  fullName: string;
  email: string;
  phoneNumber: string;
  licenseNumber: string;
  licenseExpiryDate: string;
  status: DriverStatus;
  currentVehicleId?: string;
  currentVehicleRegistration?: string;
}

export interface CreateDriverRequest {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  licenseNumber: string;
  licenseExpiryDate: string;
}

export interface UpdateDriverRequest {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  licenseExpiryDate: string;
  status: DriverStatus;
}
